src/pybind/mgr/devicehealth/module.py

```diff
@@ -631,3 +631,7 @@ def predict_all_devices(self):
                 return self.remote(plugin_name, 'predict_all_devices')
         except:
             return -1, '', 'unable to invoke diskprediction local or remote plugin'
+
+    def gather_device_report(self):
+        # write me
```

Comment:
I guess you want to implement this before merging?

Reply:
I don't want to hold this up... let's just leave it as a stub for now.